library/src/main/java/io/voucherify/android/client/module/Validation.java

```diff
@@ -0,0 +1,120 @@
+package io.voucherify.android.client.module;
+
+import java.io.IOException;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import io.reactivex.Observable;
+import io.reactivex.Scheduler;
+import io.voucherify.android.client.api.VoucherifyApi;
+import io.voucherify.android.client.callback.VoucherifyCallback;
+import io.voucherify.android.client.exception.VoucherifyError;
+import io.voucherify.android.client.model.OrderItem;
+import io.voucherify.android.client.model.VoucherResponse;
+import io.voucherify.android.client.utils.RxUtils;
+
+public class Validation extends AbsModule<Validation.ExtAsync, Validation.ExtRxJava> {
+    private final VoucherifyApi api;
+
+    private static final String CHANNEL = "android";
+
+    private final String trackingId;
+
+    private final Scheduler scheduler;
+
+    Validation(VoucherifyApi api, Scheduler scheduler, String trackingId) {
+        super(api, scheduler);
+        this.api = api;
+        this.scheduler = scheduler;
+        this.trackingId = trackingId;
+    }
+
+    public VoucherResponse validateVoucher(String code) throws IOException {
+        return validateVoucher(code, null);
+    }
+
+    public VoucherResponse validateVoucher(String code, Integer amount) throws IOException {
+        return validateVoucher(code, amount, null);
+    }
+
+    public VoucherResponse validateVoucher(String code, Integer amount, List<OrderItem> orderItems) throws IOException {
+        Map<String, String> queryParams = new LinkedHashMap<>();
+        queryParams.put("code", code);
+        if (trackingId != null) {
```

Comment:
What if we send 'null' as 'tracking_id' or 'amount'? If Retrofit doesn't send it then, we could remove this it/else checking.